File: artifacts/trading-app/src/lib/api/market.ts
import { api } from "./client";
import type { MarketOverview, StockQuote, OrderBook } from "@/types";

export const marketApi = {
  getOverview: () => api.get<MarketOverview>("/api/market/overview"),
  getQuote: (symbol: string) => api.get<StockQuote>(`/api/market/quote/${symbol}`),
  getOrderBook: (symbol: string) => api.get<OrderBook>(`/api/market/orderbook/${symbol}`),
};

File: backend/app/routers/market.py
from fastapi import APIRouter
from ..schemas.market import MarketOverview, StockQuote, OrderBook
from ..services.upstox_service import get_quote, get_order_book, get_market_overview

router = APIRouter(prefix="/api/market", tags=["market"])


@router.get("/overview", response_model=MarketOverview)
async def market_overview():
    return get_market_overview()


@router.get("/quote/{symbol}", response_model=StockQuote)
async def stock_quote(symbol: str):
    return get_quote(symbol.upper())


@router.get("/orderbook/{symbol}", response_model=OrderBook)
async def order_book(symbol: str):
    return get_order_book(symbol.upper())
